Smart resource pack parsing for file drops / opens

ryanccn opened the pull request:
This PR parses files opened via XDG, Finder, CLI, or drag and drop to check if they are resource packs, and opens a dialog to import it into an instance if it is one. Otherwise, it falls back to the original instance importing behavior in #229.

Signed-off-by: Ryan Cao <70191398+ryanccn@users.noreply.github.com>

<!--
Hey there! Thanks for your contribution.

Please make sure that your commits are signed off first.
If you don't know how that works, check out our contribution guidelines: https://github.com/PrismLauncher/PrismLauncher/blob/develop/CONTRIBUTING.md#signing-your-work
If you already created your commits, you can run `git rebase --signoff develop` to retroactively sign-off all your commits and `git push --force` to override what you have pushed already.

Note that signing and signing-off are two different things!
-->

ryanccn committed:
feat: initial support for smart resource pack parsing on file handler

Signed-off-by: Ryan Cao <70191398+ryanccn@users.noreply.github.com>

ryanccn committed:
fix resource packs and add support for texture packs

Signed-off-by: Ryan Cao <70191398+ryanccn@users.noreply.github.com>

ryanccn committed:
fix: check if packFormat is valid for resourcepack parsing

Signed-off-by: Ryan Cao <70191398+ryanccn@users.noreply.github.com>

ryanccn committed:
fix: add resource pack format for 1.19.3

Co-authored-by: Rachel Powers <508861+Ryex@users.noreply.github.com>
Signed-off-by: Ryan Cao <70191398+ryanccn@users.noreply.github.com>

ryanccn committed:
fix instance IDs and resource pack imports

Signed-off-by: Ryan Cao <70191398+ryanccn@users.noreply.github.com>

ryanccn committed:
fix: resource pack import dialog icons and add message

Signed-off-by: Ryan Cao <70191398+ryanccn@users.noreply.github.com>

ryanccn committed:
fix: add support for CLI and drag and drop

Signed-off-by: Ryan Cao <70191398+ryanccn@users.noreply.github.com>

ryanccn committed:
fix: stray include

Signed-off-by: Ryan Cao <70191398+ryanccn@users.noreply.github.com>

flowln committed:
refactor: move RP/TP validation to their respective utils

This makes it easier to validate individual resources, and allows the
logic to be used in other places in the future, if we need to.

Signed-off-by: flow <flowlnlnln@gmail.com>

flowln committed:
fix: remove commented-out assertions

They may not hold true at all times, and that's fine.

Signed-off-by: flow <flowlnlnln@gmail.com>

flowln reviewed (commented):
I've added a couple commits to take the logic you implemented into a separate thing, so that we can maybe fix #349 with it later on.

The instance names get cut off when they're too long. I believe that's the same issue we had with icon names at one point, so maybe there's something there to help fix this :p
![image](https://user-images.githubusercontent.com/9145768/202925534-f06e84c6-cc4e-450e-8f2e-864c35228296.png)

ryanccn commented:
Seems like the same issue still exists with icon names O.o

I copied `ImportResourcePackDialog` off of `IconPickerDialog` so they should have the same UI and behavior already.

ryanccn committed:
fix: text wrapping

Signed-off-by: Ryan Cao <70191398+ryanccn@users.noreply.github.com>

ryanccn committed:
Merge branch 'develop' of https://github.com/PrismLauncher/PrismLauncher into import-resource-pack-dialog-uwu

flowln committed:
fix(RPImportDialog): allow changing item size to accomodate wrapped text

Signed-off-by: flow <flowlnlnln@gmail.com>

flowln reviewed (commented):
how do you use that with CLI? `-I` open the modpack import thingy, so I guess that's not it :monocle_face: 

This looks cool. It would perhaps be cooler, though, if there was some info on the RP/TP in the dialog. Perhaps an [InfoFrame](https://github.com/PrismLauncher/PrismLauncher/blob/develop/launcher/ui/widgets/InfoFrame.h) on the top area would look good :>

Scrumplex committed:
fix: improve code readability

Signed-off-by: Sefa Eyeoglu <contact@scrumplex.net>

Scrumplex committed:
fix: check resource pack validity solely using pack format

Signed-off-by: Sefa Eyeoglu <contact@scrumplex.net>

Scrumplex committed:
Merge remote-tracking branch 'upstream/develop' into import-resource-pack-dialog-uwu

flowln committed:
fix: explicitly set scheme for local file in -I argument

Otherwise isLocalFile() will return false for local files without the
file:// thingy.

Signed-off-by: flow <flowlnlnln@gmail.com>
